src/common/exporter/word/exporters/outline.js

```diff
@@ -44,15 +45,23 @@ export default function exportOutline(state, namesMapping, doc, options) {
     const uniqueBeatTitleSelector = makeBeatTitleSelector(state)
     const title = uniqueBeatTitleSelector(state, beat.id)
     let paragraphs = [new Paragraph('')]
-    paragraphs.push(new Paragraph({ text: title, heading: HeadingLevel.HEADING_2 }))
+    if (!outlineFilter) {
+      paragraphs.push(new Paragraph({ text: title, heading: HeadingLevel.HEADING_2 }))
+    }
 
     if (options.outline.sceneCards) {
-      const cards = beatCardMapping[beat.id]
+      let cards = beatCardMapping[beat.id]
       const customAttrs = cardsCustomAttributesSelector(state)
       const sortedCards = sortCardsInBeat(beat.autoOutlineSort, cards, lines)
-      let cardParagraphs = sortedCards.flatMap((c) =>
-        card(c, linesById, namesMapping, customAttrs, doc, options)
-      )
+      let cardParagraphs = sortedCards.flatMap((c) => {
+        if (outlineFilter && !outlineFilter.includes(c.lineId)) {
+          return
+        }
+        if (outlineFilter) {
+          paragraphs.push(new Paragraph({ text: title, heading: HeadingLevel.HEADING_2 }))
```

Comment:
This side effect shouldn't happen inside of a call to `flatMap`.
Am I right in guessing that this would do the trick:
```
sortedCards.forEach((c) => {
  if (outlineFilter && !outlineFilter.includes(c.lineId))  return
  if (outlineFilter) paragraphs.push(new Paragraph({ text: title, heading: HeadingLevel.HEADING_2 }))
})

let cardParagraphs = sortedCards.flatMap...
```